### ISSUE TITLE
False positives when using the null-safe operator `?->` in an if-condition

### ISSUE THREAD

**Issue by windaishi** [kalessil/phpinspectionsea#1954]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 5.0.0.0 |
| Language level | e.g. PHP 8.2 (check project settings or composer.json)                        |

Current behaviour
--

The following code is reported with the following two issues:

Line 5: [EA] Null pointer exception may occur here. (on `$shippingMethodConfig`)
Line 6: [EA] The if-else workflow is driven by inverted conditions, consider avoiding invertions. (on `else)

```
/** @var ShippingMethodConfigEntity|null $shippingMethodConfig */
$shippingMethodConfig;

if ($shippingMethodConfig?->getAddressConfiguration()->useDefaultAddress() === false) {
    $senderAddress = $shippingMethodConfig->getAddressConfiguration()->getAddress();
} else {
    $senderAddress = $commonConfig->getSenderAddress();
}
```

Expected behaviour
--

No errors are reported. Neither is the if driven by an inverted condition (because we are comparing against `bool?`) nor can  `shippingMethodConfig` be null here.

Environment details
---

<img width="503" alt="image" src="https://github.com/user-attachments/assets/958f95b4-744f-46e3-9e83-f41d29b3017b">

**Comment by ea-inspections-team** [comment]
Thank you for reporting @windaishi!

We have multiple reports for this case and will include a fix into next release.

**Comment by ea-inspections-team** [comment]
We are pushing fix for `Line 6: [EA] The if-else workflow is driven by inverted conditions, consider avoiding invertions. (on `else)` false-positive, which greatly improves accuracy if the inspection. Thank you for reporting it @windaishi!

We are unsure how to proceed with `Line 5: [EA] Null pointer exception may occur here. (on $shippingMethodConfig)`, @kalessil could you please share your thought about the false-positive.

**Comment by kalessil** [comment]
@ea-inspections-team: agree with @windaishi, it's a false-positive. 

From a types resolving resolving perspective, the code is correct (assignment in if-body will not inject null into the result type). I'm in support of fixing this false-positive.

**Comment by ea-inspections-team** [comment]
@kalessil: thank you. Fixing this is not a trivial task and we have to plan the fix for 2024.3 (not 2024.2).

**Comment by ghbob** [comment]
Yup, still occurs.
If someone wants to disable this specific inspection:
Php Inspections (EA Extended) → Probable bugs → Null reference

probably only makes sense though if something else also checks this, e.g. PhpStan.

**Comment by ea-inspections-team** [comment]
Thank you for reporting, @windaishi! This inspection appears to be somewhat unreliable, so we are disabling it in e1263f8b0607a1f50598cc49c6dbfdcdd73f4baa for now.
